CMakeLists.txt

```diff
@@ -287,15 +297,18 @@ endif()
 
 
 #Feature/Inlinentd
-if(POLYHOOK_FEATURE_INLINENTD)
-	if (POLYHOOK_USE_EXTERNAL_ASMJIT)
-		find_library(ASMJIT_LIBRARY NAMES asmjit)
-		find_path(ASMJIT_INCLUDE_DIR NAMES asmjit/asmjit.h)
-		target_link_libraries(${PROJECT_NAME} PRIVATE ${ASMJIT_LIBRARY})
-		target_include_directories(${PROJECT_NAME} PRIVATE ${ASMJIT_INCLUDE_DIR})
-	else()
-		target_link_libraries(${PROJECT_NAME} PRIVATE $<BUILD_INTERFACE:asmjit>)
-	endif()
+if(asmjit)
+    # Avoid linking asmjit twice
+    if(NOT POLYHOOK_FEATURE_DETOURS OR POLYHOOK_USE_EXTERNAL_ASMTK)
```

Comment:
What actually happens if you link twice, does it error? Is this required?

Reply:
I am refactoring it at the moment, it will make the code a bit clearer. The problem is not really linking twice, but including it. asmtk includes the asmjit's CMakeLists.txt file directly, so we must make sure not to add_subdirectory if we already included asmtk.

Reply:
Otherwise CMake fail when executing asmjit's CMakeLists.txt the second time because of `add_library` call with the existing target.

Reply:
ah ok, the inclusions make sense!